uc_order/uc_order.entity.inc

```diff
@@ -152,27 +145,28 @@ class UcOrderProduct extends Entity {
     return 'uc_order_product';
   }
 
-  /**
-   * Implements EntityInterface::bundle().
-   */
-  public function bundle() {
-    // return $this->type;
-  }
-
   /**
    * Implements EntityInterface::label().
    */
   public function label() {
-    return $this->title;
+    // Since uc_order_product entities are derived from nodes, we return the
+    // title of its "parent" node.
+    $node = node_load($this->order_product_id);
+    if ($node) {
+    return $node->title;
```

Comment:
uc_order_product entities already have a `title` property (defined in the schema). Rather than loading the node, it's better to do

`return $this->title;`

See attached image that shows the title property.

<img width="1195" alt="Screen Shot 2022-05-30 at 9 23 45 AM" src="https://user-images.githubusercontent.com/9938978/171022845-6125dcc3-5d81-45d3-9c91-c02d1378ade1.png">

Reply:
I think in part your code may be based in the fact the title is not openly declared as a property in clas UcOrderProduct. This property is dynamically added to the class when the Controller loads (or creates) a new instance of the class. This is very common in Drupal/Backdrop entities. 

A bit of background: Drupal used stdClass instead of named classes to load core entities, and therefore the class properties could actually not be openly declared. ALL of them were added dynamically by the controller. Entity API and then Backdrop (and D8) adopted named classes to load entities (although in D7 core entities remained stdClass). When named classes were introduced, that opened the option of declaring the properties of the entity. But still, this is not required, unless the property was a custom property that was not stored as a column in the database. Otherwise Backdrop creates those properties dynamically based on the names of the columns. 

In Backdrop developers have decided to openly declare those database-based properties. But this doesn't preclude us from using a dynamic property - those are accessible as soon as the entity is loaded or created (they are all empty when the entity is first created, except for the type property in most cases).

Reply:
Thanks for explaining this. I've reverted the change.

All the machinery around entities has always been a but fuzzy to me. Your explanation, plus the related adventures in bundle caching, have been most illuminating!